culane_metric/evaluate.py

```diff
@@ -0,0 +1,205 @@
+    h, w, _ = img_shape
+    image = Polygon([(0, 0), (0, h - 1), (w - 1, h - 1), (w - 1, 0)])
+    xs = [LineString(lane).buffer(distance=width / 2., cap_style=1, join_style=2).intersection(image) for lane in xs]
+    ys = [LineString(lane).buffer(distance=width / 2., cap_style=1, join_style=2).intersection(image) for lane in ys]
+
+    ious = np.zeros((len(xs), len(ys)))
+    for i, x in enumerate(xs):
+        for j, y in enumerate(ys):
+            ious[i, j] = x.intersection(y).area / x.union(y).area
+
+    return ious
+
+
+def interp(points, n=50):
+    x = [x for x, _ in points]
+    y = [y for _, y in points]
+    tck, _ = splprep([x, y], s=0, t=n, k=min(3, len(points) - 1))
+
+    u = np.linspace(0., 1., n)
+    return np.array(splev(u, tck)).T
+
+
+def culane_metric(pred, anno, width=30, iou_threshold=0.5, unofficial=False, img_shape=LLAMAS_IMG_RES):
+    if len(pred) == 0:
+        return 0, 0, len(anno)
+    if len(anno) == 0:
+        return 0, len(pred), 0
+    interp_pred = np.array([interp(pred_lane, n=50) for pred_lane in pred])  # (4, 50, 2)
+    interp_anno = np.array([interp(anno_lane, n=50) for anno_lane in anno])  # (4, 50, 2)
+
+    if unofficial:
+        ious = continuous_cross_iou(interp_pred, interp_anno, width=width, img_shape=img_shape)
+    else:
+        ious = discrete_cross_iou(interp_pred, interp_anno, width=width, img_shape=img_shape)
+
+    row_ind, col_ind = linear_sum_assignment(1 - ious)
+    tp = int((ious[row_ind, col_ind] > iou_threshold).sum())
+    fp = len(pred) - tp
+    fn = len(anno) - tp
+    # pred_ious = np.zeros(len(pred))
+    # pred_ious[row_ind] = ious[row_ind, col_ind]
+    return tp, fp, fn
+
+
+def load_prediction(path):
+    with open(path, 'r') as data_file:
+        img_data = data_file.readlines()
+    img_data = [line.split() for line in img_data]
+    img_data = [list(map(float, lane)) for lane in img_data]
+    img_data = [[(lane[i], lane[i + 1]) for i in range(0, len(lane), 2)] for lane in img_data]
+    img_data = [lane for lane in img_data if len(lane) >= 2]
+
+    return img_data
+
+
+def load_prediction_list(label_paths, pred_dir):
+    return [load_prediction(os.path.join(pred_dir, path.replace('.json', '.lines.txt'))) for path in label_paths]
+
+
+def load_labels(label_dir):
+    label_paths = helper_scripts.get_files_from_folder(label_dir, '.json')
+    annos = [[add_ys(xs) for xs in spline_creator.get_horizontal_values_for_four_lanes(label_path) if
+              (np.array(xs) >= 0).sum() > 1]  # lanes annotated with a single point are ignored
+             for label_path in label_paths]
+    label_paths = [
+        helper_scripts.get_label_base(p) for p in label_paths
+    ]
+    return np.array(annos, dtype=object), np.array(label_paths, dtype=object)
+
+
+def eval_predictions(pred_dir, anno_dir, width=30, unofficial=True, sequential=False):
+    print('Loading annotation data ({})...'.format(anno_dir))
```

Comment:
Optional: f strings are super nice for replacing most formats
```suggestion
    print(f'Loading annotation data ({anno_dir})...')
```